frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/VerifyMfaCodeHandler.java

```diff
@@ -254,11 +266,40 @@ private void processCodeSession(
         auditService.submitAuditEvent(auditableEvent, auditContext, metadataPairs);
 
         if (errorResponse.isEmpty()) {
+            if (configurationService.isAuthenticationAttemptsServiceEnabled()
+                    && codeRequest.getMfaMethodType() == MFAMethodType.AUTH_APP) {
+                Optional<UserProfile> userProfile = userContext.getUserProfile();
+                userProfile.ifPresent(
+                        profile ->
+                                authenticationAttemptsService.deleteCount(
+                                        profile.getSubjectID(),
+                                        JourneyType.REAUTHENTICATION,
+                                        CountType.ENTER_AUTH_APP_CODE));
+            }
             mfaCodeProcessor.processSuccessfulCodeRequest(
                     IpAddressHelper.extractIpAddress(input),
                     extractPersistentIdFromHeaders(input.getHeaders()));
         }
 
+        if (errorResponse.isPresent()
+                && errorResponse.get() == ErrorResponse.ERROR_1043
+                && configurationService.isAuthenticationAttemptsServiceEnabled()
```

Comment:
Could we extract this check and have a specific check to catch where we've received a reauth request but we haven't enabled reauth?  Would make this if statement a little easier to understand.